<robot name="6link_robot">
	
	<material name="black">
		<color rgba="0 1 0 1"/>
	</material>
	<material name="white">
		<color rgba="0 0 1 1"/>
	</material>

	<link name="base_link"/>

	<link name="arm1">
		<visual>
			<geometry>
				<box size="1 .05 .05"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0.5 0 0"/>
			<material name="black"/>
		</visual>
	</link>

	<joint name="joint1" type="continuous">
		<axis rpy="0 0 0" xyz="0 0 1"/>
		<parent link="base_link"/>
		<child link="arm1"/>
	</joint>

	<link name="arm2">
		<visual>
			<geometry>
				<box size="1 .05 .05"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0.5 0 0"/>
			<material name="white"/>
		</visual>
	</link>

	<joint name="joint2" type="continuous">
		<axis rpy="0 0 0" xyz="0 0 1"/>
		<parent link="arm1"/>
		<child link="arm2"/>
		<origin rpy="0 0 0" xyz="1 0 0"/>
	</joint>
	<link name="arm3">
		<visual>
			<geometry>
				<box size="1 .05 .05"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0.5 0 0"/>
			<material name="black"/>
		</visual>
	</link>

	<joint name="joint3" type="continuous">
		<axis rpy="0 0 0" xyz="0 0 1"/>
		<parent link="arm2"/>
		<child link="arm3"/>
		<origin rpy="0 0 0" xyz="1 0 0"/>
	</joint>

	<link name="arm4">
		<visual>
			<geometry>
				<box size="1 .05 .05"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0.5 0 0"/>
			<material name="white"/>
		</visual>
	</link>

	<joint name="joint4" type="continuous">
		<axis rpy="0 0 0" xyz="0 0 1"/>
		<parent link="arm3"/>
		<child link="arm4"/>
		<origin rpy="0 0 0" xyz="1 0 0"/>
	</joint>

	<link name="arm5">
		<visual>
			<geometry>
				<box size="1 .05 .05"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0.5 0 0"/>
			<material name="black"/>
		</visual>
	</link>

	<joint name="joint5" type="continuous">
		<axis rpy="0 0 0" xyz="0 0 1"/>
		<parent link="arm4"/>
		<child link="arm5"/>
		<origin rpy="0 0 0" xyz="1 0 0"/>
	</joint>

	<link name="arm6">
		<visual>
			<geometry>
				<box size="1 .05 .05"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0.5 0 0"/>
			<material name="white"/>
		</visual>
	</link>

	<joint name="joint6" type="continuous">
		<axis rpy="0 0 0" xyz="0 0 1"/>
		<parent link="arm5"/>
		<child link="arm6"/>
		<origin rpy="0 0 0" xyz="1 0 0"/>
	</joint>


	<link name="endEffector"/>

	<joint name="ee_joint" type="fixed">
		<parent link="arm6"/>
		<child link="endEffector"/>
		<origin rpy="0 0 0" xyz="1 0 0"/>
	</joint>

</robot>


